Saule/Serialization/ResourceSerializer.cs

```diff
@@ -235,24 +248,39 @@ private JObject SerializeRelationships(ResourceGraphNode node)
 
             foreach (var kv in node.Relationships)
             {
+                var relationship = kv.Value.Relationship;
+
                 var item = new JObject();
 
                 var data = SerializeRelationshipData(kv.Value);
 
                 var relationshipId = default(string);
 
                 if (data != null
-                    && kv.Value.Relationship.Kind == RelationshipKind.BelongsTo
+                    && relationship.Kind == RelationshipKind.BelongsTo
                     && kv.Value.SourceObject != null)
                 {
                     relationshipId = (string)data["id"];
                 }
 
                 var links = new JObject();
-                AddUrl(links, "self", _urlBuilder.BuildRelationshipPath(node.Resource, node.Key.Id.ToString(), kv.Value.Relationship));
-                AddUrl(links, "related", _urlBuilder.BuildRelationshipPath(node.Resource, node.Key.Id.ToString(), kv.Value.Relationship, relationshipId));
+                var self = _urlBuilder.BuildRelationshipPath(node.Resource, node.Key.Id.ToString(), relationship);
+                var related = _urlBuilder.BuildRelationshipPath(node.Resource, node.Key.Id.ToString(), relationship, relationshipId);
 
-                item["links"] = links;
+                if (!string.IsNullOrEmpty(self) && relationship.LinkType.HasFlag(LinkType.Self))
```

Comment:
I wrote down some thoughts on this below. Please let me know what you think.

It's most likely that one of the two will just be `All`. I expect the most common scenario is to just override something in one place (i.e. either the relationship or on the model, depending which best maps to the task.

This implies to me that we should take the *most restrictive* of the options. Examples:

- Person -> Company (Job relationship). Person and company both have link type `All`, the relationship has link type `Related`. When serializing individual person or company records, we should serialize all links. When serializing the job relationship, we should serialize only `related` links.
- Person -> Company (Job relationship). Person has link type `All`, company has `Related`. The relationship has link type `All`. When serializing individual person records, we should serialize all links. For company records, we should not serialize anything (because there's no related links in that case). For the job relationship, we should serialize only `related` links.

One case where I'm not sure:
- Person -> Company (Job relationship). Person and company both have link type `None`. The relationship has link type `Related`. When serializing individual records, we should serialize no links. For the job relationship, what should we do? It sort of makes sense to serialize no links, but then again, the user explicitly set the link type to `Related` so they probably want those.

I guess the solution would be something like this:

1. When not serializing relationships, there is only one option so this should be simple.
2. Otherwise, if one of the options is `All`, take the other option.
3. In all other cases, use the logical 'or' of the options.

Examples of this:

- `All` and `None` -> serialize none
- `All` and `Related` -> serialize related
- `Related` and `None` -> serialize none
- `Related` and `Self` -> serialize related & self only (where they make sense)

Of course, we always have the option to just ignore one of the two:

1. When serializing the `related` hash, use the relationship link type
2. When serializing models (i.e. the top level `data` or `included` hash), use the resource link type

This is simple (I believe this is what the code in this PR does now) and seems the easiest to document.